EDITORIAL CLOSING: XMPP Newsletter December 2022 & January 2023 (#1232)
xsf/xmpp.org: content/blog/2023-02-05-newsletter.md

---
title: The XMPP Newsletter December 2022 & January 2023
date: 2023-02-05
categories: ['Newsletter']
---

Welcome to the XMPP Newsletter, great to have you here again! This issue covers the month of December 2022 & January 2023. 
This is the first release of this year and we have after a well-deserved winter break! Many thanks to all our readers and all contributors!

Like this newsletter, many projects and their efforts in the XMPP community are a result of people’s voluntary work. If you are happy with the services and software you may be using, especially throughout the current situation, please consider saying thanks or help these projects! Interested in supporting the Newsletter team? Read more at the bottom.

> ### Newsletter Contributors & Translations

Author: Ok

[A paper](https://ieeexplore.ieee.org/document/9968404) was presented on IECON2022, and just recently published. Based on XMPP.

[JMP's December Newsletter](https://blog.jmp.chat/b/december-newsletter-2022) talks about their new co-op structure, an important security fix in the Cheogram Android app, and a review of the highlights of 2022.  [JMP's January Newsletter](https://blog.jmp.chat/b/january-newsletter-2023) mentions another new release including a new [XMPP Threaded Chat UI](https://kumi.tube/w/1LQQp5Uia4u8Pdojxen1y8) and new payment options for the service. Last but not least: [Writing a Chat Client from Scratch](https://blog.jmp.chat/b/2022-chat-client-from-scratch) in Ruby.

Snikket informs about the [F-Droid security warnings](https://snikket.org/blog/fdroid-security-warning/) and their [Security update](https://snikket.org/blog/fdroid-security-update/). The issues has been solved already.

PGPainless: [Use Any SOP Binary With SOP-Java and External-SOP](https://blog.jabberhead.tk/2023/01/13/use-any-sop-binary-with-sop-java-and-external-sop/)

From the ignite relatime community:

> - [Spark 3.0.1 Released](https://discourse.igniterealtime.org/t/spark-3-0-1-released/92353)

Author: Let's keep it anyway

> - [Denial of Service Vulnerability in Smack 4.4 if XMPPTCPConnection is used with StAX](https://discourse.igniterealtime.org/t/denial-of-service-vulnerability-in-smack-4-4-if-xmpptcpconnection-is-used-with-stax/92314)

Reviewer: ```suggestion
- A Denial of Service Vulnerability in Smack 4.4 [has been fixed](https://discourse.igniterealtime.org/t/denial-of-service-vulnerability-in-smack-4-4-if-xmpptcpconnection-is-used-with-stax/92314)
```

> - [New Mastodon service for the Ignite Realtime community!](https://discourse.igniterealtime.org/t/new-mastodon-service-for-the-ignite-realtime-community/92389)

Reviewer: ```suggestion
- [The Ignite Realtime community now offers a Mastodon service](https://discourse.igniterealtime.org/t/new-mastodon-service-for-the-ignite-realtime-community/92389)
```

> - [Help us translate Spark and Openfire!](https://discourse.igniterealtime.org/t/help-us-translate-spark-and-openfire/92399)

Reviewer: ```suggestion
- [The community started a call for help with translating Spark and Openfire](https://discourse.igniterealtime.org/t/help-us-translate-spark-and-openfire/92399)
```

> [EU NIS 2 directive: does it apply to anyone who provides free online digital services?](https://notes.nicfab.it/en/posts/nis2/)

Reviewer: ```suggestion
nicfab writes about the new [EU NIS 2 directive: does it apply to anyone who provides free online digital services?](https://notes.nicfab.it/en/posts/nis2/)
```

[Gajim 1.6.0](https://gajim.org/post/2023-01-07-gajim-1.6.0-released/) and [1.6.1](https://gajim.org/post/2023-01-14-gajim-1.6.1-released/) have been released. For Gajim 1.6, developers ported Gajim’s and python-nbxmpp’s underlying HTTP library to libsoup3. Also, audio previews now look nicer and allow for more control (playback speed, jumping). Emoji rendering on MacOS has been repaired and many fixes have been implemented under the hood.

Libervia has been granted a [NLnet NGI Assure Fund for implementation of Audio/Video Calls](https://nlnet.nl/project/Libervia-AV/).

[Moxxy 0.4.0](https://codeberg.org/moxxy/moxxy/releases/tag/v0.4.0) and [0.4.1](https://codeberg.org/moxxy/moxxy/releases/tag/v0.4.1) have been released. These two releases bring stickers, reactions, voice message recording and various improvements and fixes to the UI and XMPP code.

> [Spark 3.0.1 has been released](https://github.com/igniterealtime/Spark/releases), which brings some bug fixes

Reviewer: ```suggestion
[Spark 3.0.1 has been released](https://github.com/igniterealtime/Spark/releases), which brings some bug fixes.
```

### Servers

> [ejabberd 23.01 has been released](https://www.process-one.net/blog/ejabberd-23-01/) Almost three months after the previous release, ejabberd 23.01 includes many bug fixes, several improvements and some new features. A new module, mod_mqtt_bridge, can be used to replicate changes to MQTT topics between local and remote servers. Supported XEPs are now automatically gathered from source code documentation in order to provide a DOAP file for further processing on [xmpp.org/servers/ejabberd](https://xmpp.org/software/servers/ejabberd/).

Reviewer: ```suggestion
[ejabberd 23.01 has been released](https://www.process-one.net/blog/ejabberd-23-01/). Almost three months after the previous release, ejabberd 23.01 includes many bug fixes, several improvements and some new features. A new module, mod_mqtt_bridge, can be used to replicate changes to MQTT topics between local and remote servers. Supported XEPs are now automatically gathered from source code documentation in order to provide a DOAP file for further processing on [xmpp.org/servers/ejabberd](https://xmpp.org/software/servers/ejabberd/).
```

> [Prosody 0.12.2 has been released](https://blog.prosody.im/prosody-0.12.2-released/) and [Prosody 0.11.14 released](https://blog.prosody.im/prosody-0.11.14-released/), which contains allowance U+7F when constructing stanzas, bug fixes and improvements.

Reviewer: ```suggestion
[Prosody 0.12.2](https://blog.prosody.im/prosody-0.12.2-released/) and [Prosody 0.11.14](https://blog.prosody.im/prosody-0.11.14-released/) have been released.  The 0.12 release brings bug fixes and improvements.
```

> [Tigase XMPP Server 8.3 has been released](https://tigase.org/blog/tigase-xmpp-server-8.3.0/) -  it includes Tigase MEET - Tigase's take at SFU (Selective Forwarding Unit) group video-calls, support for XEP-0440 SASL Channel Binding Type Capability as well as initial, preview support for SASL2 and Bind2 and lots of usual fixes.

Reviewer: ```suggestion
[Tigase XMPP Server 8.3 has been released](https://tigase.org/blog/tigase-xmpp-server-8.3.0/). Tt includes Tigase MEET - Tigase's take at SFU (Selective Forwarding Unit) group video-calls, support for XEP-0440 SASL Channel Binding Type Capability as well as initial, preview support for SASL2 and Bind2 and lots of usual fixes.
```

Daniel Gultsch's [compliance.conversations.im](https://compliance.conversations.im/) [will be discontinued](https://gultsch.social/@daniel/109653873511471850). This online compliance checker allowed server administrators and users to identify the XEPs/features supported by a provider, and it also allowed to check XEP support over time. This service helped users to check whether they are creating a future-proof account with a provider. Due to difficulties in maintaining the software, the service will be shut down in the near future.

[QXmpp 1.5](https://github.com/qxmpp-project/qxmpp/releases/tag/v1.5.0) has been released! The Qt-/C++-based XMPP library supports OMEMO 2, Stanza Content Encryption and Automatic Trust Management now. Much more data than with the old OMEMO versions can be end-to-end encrypted. For example, files including thumbnails are shared encrypted via Stateless File Sharing. Furthermore, Message Reactions is supported and QXmpp's PubSub implementation is almost feature-complete.

## Extensions and specifications

> Developers and other standards experts from around the world collaborate on these extensions, developing new specifications for emerging practices, and refining existing ways of doing things. Proposed by anybody, the particularly successful ones end up as Final or Active - depending on their type - while others are carefully archived as Deferred. This life cycle is described in [XEP-0001](https://xmpp.org/extensions/xep-0001.html), which contains the formal and canonical definitions for the types, states, and processes. [Read more about the standards process](https://xmpp.org/about/standards-process.html). Communication around Standards and Extensions happens in the [Standards Mailing List](https://mail.jabber.org/mailman/listinfo/standards) ([online archive](https://mail.jabber.org/pipermail/standards/)).

Reviewer: ```suggestion
The XMPP Standards Foundation develops extensions to XMPP in its [XEP series](https://xmpp.org/extensions/) in addition to [XMPP RFCs](https://xmpp.org/rfcs/).

Developers and other standards experts from around the world collaborate on these extensions, developing new specifications for emerging practices, and refining existing ways of doing things. Proposed by anybody, the particularly successful ones end up as Final or Active - depending on their type - while others are carefully archived as Deferred. This life cycle is described in [XEP-0001](https://xmpp.org/extensions/xep-0001.html), which contains the formal and canonical definitions for the types, states, and processes. [Read more about the standards process](https://xmpp.org/about/standards-process.html). Communication around Standards and Extensions happens in the [Standards Mailing List](https://mail.jabber.org/mailman/listinfo/standards) ([online archive](https://mail.jabber.org/pipermail/standards/)).
```

> xmpp.org features [a page about XMPP RFCs](https://xmpp.org/rfcs/) as well.

Reviewer: ```suggestion
```

- No Last Call this month.

### Stable

- No XEP moved to stable this month.
  
### Deprecated

- No XEP deprecated this month.

### Call for Experience

A Call For Experience - like a Last Call, is an explicit call for comments, but in this case it's mostly directed at people who've implemented, and ideally deployed, the specification. The Council then votes to move it to Final.

-   No Call for Experience this month.

## Spread the news!

Please share the news on other networks:

* [Mastodon](https://fosstodon.org/@xmpp/)
* [Twitter](https://twitter.com/xmpp)
* [YouTube](https://www.youtube.com/channel/UCf3Kq2ElJDFQhYDdjn18RuA)
* [Lemmy instance](https://community.xmpp.net/)
* [Reddit](https://www.reddit.com/r/xmpp/)
* [LinkedIn](https://www.linkedin.com/company/xmpp-standards-foundation/)

{{< newsletter-subscribe >}}

Also check out our [RSS Feed](https://xmpp.org/feeds/all.atom.xml)!

Looking for job offers or want to hire a professional consultant for your XMPP project? Visit our [XMPP job board](https://xmpp.work/).

## Help us to build the newsletter

> This XMPP Newsletter is produced collaboratively by the XMPP community. Each month's newsletter issue is drafted in this [simple pad](https://yopad.eu/p/xmpp-newsletter-365days). At the end of each month, the pad's content is merged into the [XSF Github repository](https://github.com/xsf/xmpp.org/milestone/3). We are always happy to welcome contributors. Do not hesitate to join the discussion in our [Comm-Team group chat (MUC)](xmpp:commteam@muc.xmpp.org?join) and thereby help us sustain this as a community effort. You have a project and want to spread the news? Please consider sharing your news or events here, and promote it to a large audience.

Reviewer: ```suggestion
This XMPP Newsletter is produced collaboratively by the XMPP community. Each month's newsletter issue is drafted in this [simple pad](https://pad.nixnet.services/oHnY_ZvLT8SoFyCqIC2ung). At the end of each month, the pad's content is merged into the [XSF Github repository](https://github.com/xsf/xmpp.org/milestone/3). We are always happy to welcome contributors. Do not hesitate to join the discussion in our [Comm-Team group chat (MUC)](xmpp:commteam@muc.xmpp.org?join) and thereby help us sustain this as a community effort. You have a project and want to spread the news? Please consider sharing your news or events here, and promote it to a large audience.
```
